### DISCUSSION TITLE
How to set setValueLengthLimit for HTTPRequest when using HTTPServer?

### DISCUSSION THREAD

**Discussion by starquell** [pocoproject/poco#4051] [Q&A]
Is there an option to set header value length limit for every request when using HTTPServer? HTTPRequest has method setValueLengthLimit, but I can`t use it because HTTPRequest created and header are parsed in Poco internal functions. 
Why there is no setValueLengthLimit in HTTPServerParams class?

**Comment by CyberCMRX** [comment]
We have been running into the same problem! 
We have been working with the Poco Library for almost 20 years now and were big fans of it. But now Poco has embarrassed us with the customer (more than 6000 users use the software). The customer packs all of a user's rights into a value and sends these in the header with a request to the server (for a few bosses the length of the value exceeds 8192 characters, causing error 400). 
With the Apache server, the customer could simply configure the maximum size of the header. With Poco, there is apparently no provision for configuration. Or does anyone know when and where setValueLengthLimit() should be called in the server framework???

Otherwise we can only think of setting DFL_VALUE_LENGTH_LIMIT in the code and recompiling the library (not easy to configure using vcpkg).
Does anybody know how to solve this problem by using the standard poco features properly?
